### PR TITLE
Metaclient: Drop Hadoop 2 support

### LINKED ISSUE #6201: Drop Hadoop2 support from Spark metadata client

### PR DESCRIPTION
**Pull request by johnnyaug**
Resolves #6201.

### CODE REVIEW

**Review by arielshaqed — APPROVED** [review]
You deleted all my horrible build.sbt hacks!  I will never again convince anyone that I am an 3133+ SBT h4xx0r!!

Thanks, this is truly lovely.  spark/core/src/main/hadoop{2,3} are now empty, right?

File: clients/spark/build.sbt [hunk 1]
```diff
@@ -1,102 +1,54 @@
-import build.BuildType
-
-lazy val baseName = "lakefs-spark"
 lazy val projectVersion = "0.10.0"
 lazy val hadoopVersion = "3.2.1"
 
-ThisBuild / isSnapshot := false
+ThisBuild / isSnapshot := true
```
Comment (nopcoder) [inline clients/spark/build.sbt:4]: Not expert on sbt - but do we need to mark this one always as `true` it is not set based on how we publish?

**Review by nopcoder — APPROVED** [review]
LGTM - just question about isSnapshot

File: clients/spark/build.sbt (hunk 1 above)
Comment (johnnyaug) [reply clients/spark/build.sbt:4]: It was just an experiment, good catch

**Pull request merged**